src/main/kotlin/org/opensearch/commons/alerting/action/GetMonitorResponse.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.commons.alerting.action
+
+import org.opensearch.commons.alerting.model.Monitor
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._ID
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._PRIMARY_TERM
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._SEQ_NO
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._VERSION
+import org.opensearch.commons.notifications.action.BaseResponse
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.ToXContentFragment
+import org.opensearch.core.xcontent.XContentBuilder
+import java.io.IOException
+
+class GetMonitorResponse : BaseResponse {
+    var id: String
+    var version: Long
+    var seqNo: Long
+    var primaryTerm: Long
+    var monitor: Monitor?
+    var associatedWorkflows: List<AssociatedWorkflow>?
+
+    constructor(
+        id: String,
+        version: Long,
+        seqNo: Long,
+        primaryTerm: Long,
+        monitor: Monitor?,
+        associatedCompositeMonitors: List<AssociatedWorkflow>?,
+    ) : super() {
+        this.id = id
+        this.version = version
+        this.seqNo = seqNo
+        this.primaryTerm = primaryTerm
+        this.monitor = monitor
+        this.associatedWorkflows = associatedCompositeMonitors ?: emptyList()
+    }
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        id = sin.readString(), // id
+        version = sin.readLong(), // version
+        seqNo = sin.readLong(), // seqNo
+        primaryTerm = sin.readLong(), // primaryTerm
+        monitor = if (sin.readBoolean()) {
+            Monitor.readFrom(sin) // monitor
+        } else null,
+        // TODO - find correct solution for this instead of setting to null
+        associatedCompositeMonitors = null
```

Comment:
Original code is
```
associatedCompositeMonitors = sin.readList((AssociatedWorkflow)::readFrom)
```

Getting this exception with the original code
```
[2024-02-10T14:49:27,254][ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [integTest-0] uncaught exception in thread [DefaultDispatcher-worker-8 @coroutine#8]
java.io.EOFException: null
        at org.opensearch.core.common.io.stream.InputStreamStreamInput.readBytes(InputStreamStreamInput.java:83) ~[opensearch-core-2.11.1-SNAPSHOT.jar:2.11.1-SNAPSHOT]
        at org.opensearch.core.common.io.stream.StreamInput.readString(StreamInput.java:482) ~[opensearch-core-2.11.1-SNAPSHOT.jar:2.11.1-SNAPSHOT]
        at org.opensearch.commons.alerting.action.GetMonitorResponse$AssociatedWorkflow.<init>(GetMonitorResponse.kt:114) ~[?:?]
        at org.opensearch.commons.alerting.action.GetMonitorResponse$AssociatedWorkflow$Companion.readFrom(GetMonitorResponse.kt:122) ~[?:?]
        at org.opensearch.core.common.io.stream.StreamInput.readCollection(StreamInput.java:1209) ~[opensearch-core-2.11.1-SNAPSHOT.jar:2.11.1-SNAPSHOT]
        at org.opensearch.core.common.io.stream.StreamInput.readList(StreamInput.java:1160) ~[opensearch-core-2.11.1-SNAPSHOT.jar:2.11.1-SNAPSHOT]
        at org.opensearch.commons.alerting.action.GetMonitorResponse.<init>(GetMonitorResponse.kt:54) ~[?:?]
        at org.opensearch.commons.alerting.AlertingPluginInterface$getMonitor$1.invoke(AlertingPluginInterface.kt:313) ~[?:?]
        at org.opensearch.commons.alerting.AlertingPluginInterface$getMonitor$1.invoke(AlertingPluginInterface.kt:311) ~[?:?]
        at org.opensearch.commons.alerting.AlertingPluginInterface$wrapActionListener$1.onResponse(AlertingPluginInterface.kt:373) ~[?:?]
        at org.opensearch.commons.alerting.AlertingPluginInterface$wrapActionListener$1.onResponse(AlertingPluginInterface.kt:371) ~[?:?]
        at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:113) ~[opensearch-2.11.1-SNAPSHOT.jar:2.11.1-SNAPSHOT]
        at org.opensearch.action.support.TransportAction$1.onResponse(TransportAction.java:107) ~[opensearch-2.11.1-SNAPSHOT.jar:2.11.1-SNAPSHOT]
        at org.opensearch.alerting.transport.TransportGetMonitorAction$doExecute$1$1$onResponse$2.invokeSuspend(TransportGetMonitorAction.kt:134) ~[opensearch-alerting-2.11.1.0-SNAPSHOT.jar:2.11.1.0-SNAPSHOT]
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) [kotlin-stdlib-1.8.21.jar:1.8.21-release-380(1.8.21)]
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233) [kotlinx-coroutines-core-1.1.1.jar:?]
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594) [kotlinx-coroutines-core-1.1.1.jar:?]
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60) [kotlinx-coroutines-core-1.1.1.jar:?]
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742) [kotlinx-coroutines-core-1.1.1.jar:?]
```

If anyone knows why, that would be helpful